File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people = sorted(people)
    
    answer = 0
    start, end = 0, len(people) - 1
    while start <= end:
        answer += 1
        if people[start] + people[end] <= limit:
            start += 1
        end -= 1
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        
        int answer = 0;
        int start = 0; int end = people.length - 1;
        while (start <= end) {
            answer++;
            if (people[start] + people[end] <= limit) {
                start++;
            }
            end--;
        }
        return answer;
    }
}